src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -1344,10 +1344,101 @@ void PerimeterGenerator::process_arachne()
         //    wall_0_inset = 0.5 * (ext_perimeter_width + this->perimeter_flow.scaled_width() - ext_perimeter_spacing -
         //                           perimeter_spacing);
 
+        std::vector<Arachne::VariableWidthLines> out_shell;
```

Comment:
Do you think if we can extract this block into a new function?

Reply:
I think we can, and even make it so that the classic generator can use the same function as well since this part was essentially the same code copied from it. Do you want me to do that now? Or wait until this has been thoroughly tested in case some major changes will be needed?

Reply:
Yes, that would be great.
It's better to do that now so that we can test them in one go. (in one beta release)

Reply:
Done